### PR TITLE
[bf16] pten matmul cuda kernel support bf16

### PR DESCRIPTION
**Pull request by zhiqiu**
### PR types

New features 
### PR changes

OPs
### Describe

[bf16] pten matmul and matmul_grad cuda kernel support bf16

image

image

### CODE REVIEW

File: paddle/pten/kernels/empty_kernel.cc [hunk 1]
```diff
@@ -94,6 +94,7 @@ PT_REGISTER_KERNEL(empty_like,
                    int64_t,
                    bool,
                    paddle::platform::float16,
+                   paddle::platform::bfloat16,
```
Comment (chenwhql) [inline paddle/pten/kernels/empty_kernel.cc:97]: 这里类型建议都使用pten::dtype::xxx

**Review by qili93 — APPROVED** [review]
LGTM

**Review by chenwhql — APPROVED** [review]

**Review by zhangbo9674 — APPROVED** [review]
LGTM

**Review by ZzSean — APPROVED** [review]
LGTM for op benchmark

**Pull request merged**